packages/flutter_driver/lib/src/extension/internal_wait.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/widgets.dart';
+
+import '../common/wait.dart';
+
+/// Base class for a condition that can be waited upon.
+abstract class WaitCondition {
+  /// Gets the current status of the [condition], executed in the context of the
+  /// Flutter app:
+  ///
+  /// * True, if the condition is satisfied.
+  /// * False otherwise.
+  ///
+  /// The future returned by [wait] will complete when this [condition] is
+  /// fulfilled.
+  bool get condition;
+
+  /// Returns a future that completes when [condition] turns true.
+  Future<void> wait();
+}
+
+/// A condition that waits until no transient callbacks are scheduled.
+class InternalNoTransientCallbacksCondition implements WaitCondition {
```

Comment:
Could all these "internal" classes be private?

Reply:
Yep, updated them to private.